src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -203,6 +205,10 @@ public void visitToken(DetailAST ast) {
                     visitMethodDef(ast);
                 }
                 break;
+            case TokenTypes.TYPE_UPPER_BOUNDS:
+            case TokenTypes.TYPE_ARGUMENT:
+                visitGenericArgumentDef(ast);
```

Comment:
I suspect this can result in violation message duplication in a case like `Foo<? extends Foo>`. Please add this case as an input.

Reply:
Why will be need not duplication for `Foo<? extends Foo>?`

Reply:
In violation message the different position.